ui/page/components/items_scroll.go

```diff
@@ -67,47 +70,48 @@ func (s *Scroll[T]) FetchScrollData(isReverse bool, window app.WindowNavigator)
 		s.offset -= s.pageSize
 	}
 	s.mu.Unlock()
-
-	s.fetchScrollData(isReverse, window)
+	// set isReverse to default false as callers of this method are not
+	// perform a reverse scroll action
+	s.fetchScrollData(false, isReset, window)
 }
 
 // fetchScrollData fetchs the scroll data and manages data returned depending on
 // on the up or downwards scrollbar movement. Unless the data fetched is less than
 // the page, all the old data is replaced by the new fetched data making it
 // easier and smoother to scroll on the UI. At the end of the function call
 // a window reload is triggered.
-func (s *Scroll[T]) fetchScrollData(isReverse bool, window app.WindowNavigator) {
+func (s *Scroll[T]) fetchScrollData(isReverse, isReset bool, window app.WindowNavigator) {
 	s.mu.Lock()
 
+	if isReset {
+		// resets the values for use on the next iteration.
+		s.resetList()
```

Comment:
Are the reset values used on the next iteration or right away?

Reply:
Right away. Reason is, most callers of the method needing a reset, either performed a filter change or completed sync.